src/docs/cookbook/testing/integration/introduction.md

```diff
@@ -134,187 +108,102 @@ class _MyHomePageState extends State<MyHomePage> {
+            (WidgetTester tester) async {
+          app.main();
+          await tester.pumpAndSettle();
 
-    test('starts at 0', () async {
-      // Use the `driver.getText` method to verify the counter starts at 0.
-      expect(await driver.getText(counterTextFinder), "0");
-    });
+          // Verify the counter starts at 0.
+          expect(find.text('0'), findsOneWidget);
 
-    test('increments the counter', () async {
-      // First, tap the button.
-      await driver.tap(buttonFinder);
+          // Finds the floating action button to tap on.
+          final Finder fab = find.byTooltip('Increment');
 
-      // Then, verify the counter text is incremented by 1.
-      expect(await driver.getText(counterTextFinder), "1");
-    });
+          // Emulate a tap on the floating action button.
+          await tester.tap(fab);
 
-    test('increments the counter during animation', () async {
-      await driver.runUnsynchronized(() async {
-        // First, tap the button.
-        await driver.tap(buttonFinder);
+          // Trigger a frame.
+          await tester.pumpAndSettle();
 
-        // Then, verify the counter text is incremented by 1.
-        expect(await driver.getText(counterTextFinder), "1");
-      });
-    });
+          // Verify the counter increments by 1.
+          expect(find.text('1'), findsOneWidget);
+        });
   });
 }
 ```
 
-By default, `flutter_driver` waits until there are no pending frames,
-and tests similar to the example above fail with a timeout if,
-for example, you have a continuous animation running.  In that case, wrap
-the driver actions in `runUnsynchronized` as follows:
-
-<?code-excerpt "test_driver/app_test.dart (Unsynchronized)"?>
-```dart
-test('increments the counter during animation', () async {
-  await driver.runUnsynchronized(() async {
-    // First, tap the button.
-    await driver.tap(buttonFinder);
-
-    // Then, verify the counter text is incremented by 1.
-    expect(await driver.getText(counterTextFinder), "1");
-  });
-});
-```
-
-### 6. Run the tests
+### 5. Run the integration test
 
-Now that you have an instrumented app _and_ a test suite,
-run the tests. The process of running the integration
-tests varies depending on the platform you are testing
-against. You can test against a mobile platform or the web.
+The process of running the integration tests varies depending on the platform
+you are testing against. You can test against a mobile platform or the web.
 
-#### 6a. Mobile
+#### 5a. Mobile
 
-To test on iOS or Android,
-launch an Android Emulator, iOS Simulator,
-or connect your computer to a real iOS / Android device.
-
-Then, run the following command from the root of the project:
+To test on a real iOS / Android device, first connect the device and run the
+following command from the root of the project:
 
 ```shell
-flutter drive --target=test_driver/app.dart
+flutter test integration_test/app_test.dart
```

Comment:
We can also mention that `flutter test integration_test` will run all the tests in that directory as integration tests (just like `flutter test`)